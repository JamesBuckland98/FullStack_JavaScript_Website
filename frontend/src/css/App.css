.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

ul {
  width: 100%;
  height: 100%;
  margin-bottom: 0px;
  padding: 0px;
  text-align: start;
  list-style-type: none;
  background-color: white;
  color: black;
  border: 2px solid #09D3AC;
  border-bottom: none;
}

form {
  width: 100%;
}
input {
  width: 100%;
}
