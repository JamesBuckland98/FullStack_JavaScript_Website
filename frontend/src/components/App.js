import '../css/App.css';
import Login from './Login';
import SignUp from './SignUp';
import logo from '../assets/logo.jpg';
import React, { Component } from 'react';
import MessageInput from './MessageInput';
import { getChatHistory, socket } from '../api/chatApi';

class App extends Component {

	constructor(props) {
		super(props)
		this.state = {
			messages: [],
			username: '',
			islogin: true,
			isSignUp: false,
			isChat: false,
		}
		this.setUsername = this.setUsername.bind(this);
		this.switchBetweenComponents = this.switchBetweenComponents.bind(this);
	}

	componentDidMount() {
		// Sets a socket listener as soon as the component is loaded
		socket.on('new message', newMessage => {
			var newChatHistory = [...this.state.messages, newMessage]
			this.setState({
				messages: newChatHistory,
				newMessage : newMessage
			})
		})
		// Retrieves chat history from database as soon as the component is loaded
		getChatHistory((err, response) => {
			var newChatHistory = [...this.state.messages]
			response.forEach(message => {
				newChatHistory.push(message)
			});
			this.setState({
				messages: newChatHistory
			})
		});
	}

	/**
	 * Set the username to be used between components 
	 * @param String username
	 */
	setUsername(username) {
		this.setState({
			username: username
		})
	}

	/**
	 * Switch states so components can be rendered
	 * @param String component
	 */
	switchBetweenComponents(component) {
		switch (component) {
			case 'login':
				this.setState(prevState => ({
					islogin: !prevState.islogin,
				}));
				break;
			case 'sign up':
				this.setState(prevState => ({
					isSignUp: !prevState.isSignUp
				}));
				break;
			case 'chat':
				this.setState(prevState => ({
					isChat : !prevState.isChat
				}));
				break;
			default:
				break;
		}
	}

	render() {
		return (
			<div className="App">
				<header className="App-header">
					<h1> { process.env.REACT_APP_COMPANY_TITLE }</h1>
					<img src={logo} className="App-logo" alt="logo" />
					{ 
						this.state.islogin ?
						<Login
							setUsername = { this.setUsername }
							switch = { this.switchBetweenComponents }
						/> :
						null
					}
					{ 
						this.state.isSignUp ?
						<SignUp
							switch = { this.switchBetweenComponents }
						/> :
						null
					}
					{ 
						this.state.isChat ?
						<MessageInput
							emitMessage = { this.emitMessage } 
							username = { this.state.username } 
							newMessage = { this.state.newMessage }
							messages = { this.state.messages }
						/> :
						null
					}
				</header>
			</div>
		);	
	}
}

export default App;
