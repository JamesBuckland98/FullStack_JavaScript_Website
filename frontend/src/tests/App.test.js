import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../components/App';

it('should render App correctly', () => {
	const wrapper = shallow(<App/>);
	expect(toJson(wrapper)).toMatchSnapshot();
});
