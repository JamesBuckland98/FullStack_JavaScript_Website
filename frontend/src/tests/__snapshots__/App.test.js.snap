// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
       
      Sappo
    </h1>
    <img
      alt="logo"
      className="App-logo"
      src="logo.jpg"
    />
    <Login
      setUsername={[Function]}
      switch={[Function]}
    />
  </header>
</div>
`;
